src/app.ts

```diff
@@ -45,10 +55,30 @@ async function app() {
 
   logger.info(`Running SUSHI ${getVersion()}`);
 
-  // Check that input folder is specified
+  // If no input folder is specified, set default to current directory
   if (!input) {
-    logger.error('Missing path to FSH definition folder.');
-    program.help();
+    input = path.join('.');
+    logger.info('path-to-fsh-defs defaulted to current working directory');
+  }
+
+  // If a fsh subdirectory is detected, we are in an IG Publisher context
+  const isUsingFshDir = path.parse(input).base === 'fsh';
+
+  const fshSubdirectoryPath = path.join(input, 'fsh');
+  const hasFshDir = fs.existsSync(fshSubdirectoryPath);
+  const shouldUseFshDir = hasFshDir && input !== fshSubdirectoryPath;
+
+  const isIgPubContext = isUsingFshDir || shouldUseFshDir;
+  if (isIgPubContext) {
+    logger.info(
+      'Current FSH tank conforms to an IG Publisher context. Default input and output will be adjusted accordingly.'
+    );
+  }
+
+  // Use fsh/ subdirectory if not already specified and present
+  if (shouldUseFshDir) {
+    input = path.join(input, 'fsh');
+    logger.info('fsh/ subdirectory detected and add to input path');
```

Comment:
I feel like the above 15 lines of code or so can be simplified some... I'm not _sure_, but it seems like if we check the subfolder and modify the input first, then we can check if the path base is fsh after that? Then we'd have fewer similar variables to create and track.